modules/lfs/http_client.go

```diff
@@ -219,3 +193,65 @@ func (c *HTTPClient) performOperation(ctx context.Context, objects []Pointer, dc
 
 	return nil
 }
+
+// createRequest creates a new request, and sets the Content-Type and Accept headers.
+func createRequest(ctx context.Context, method, url, contentType string, headers map[string]string, body io.Reader) (*http.Request, error) {
+	log.Trace("createRequest: %s", url)
+	req, err := http.NewRequestWithContext(ctx, method, url, body)
+	if err != nil {
+		log.Error("Error creating request: %v", err)
+		return nil, err
+	}
+
+	for key, value := range headers {
+		req.Header.Set(key, value)
+	}
+	if req.Header.Get("Content-Type") == "" {
+		req.Header.Set("Content-Type", contentType)
+	}
```

Comment:
I think it could get rid of the `contentType` parameter.

Let callers pass the `headers` correctly, or check&add the header after the `createRequest` call.